### ISSUE TITLE
InvalidRouteTableID.NotFound: The routeTable ID 'rtb-xxxxxx' does not exist

### ISSUE THREAD

**Issue by brikis98** [hashicorp/terraform#7104]
### Terraform Version
Terraform v0.6.16

### Affected Resource(s)

- `aws_route_table`
- `aws_route_table_association`

### Terraform Configuration Files

We have a relatively complicated VPC setup with a number of subnets, route tables, NAT gateways, etc. It's hard for me to tell exactly which part is causing the issue, but I'm guessing it's this part:

```hcl
resource "aws_route_table" "private-persistence" {
    count = "${length(split(",", var.aws_availability_zones))}"
    vpc_id = "${aws_vpc.main.id}"
    tags {
        Name = "${var.vpc_name}-private-persistence-${count.index}"
    }
}

resource "aws_route_table_association" "private-persistence" {
    count = "${length(split(",", var.aws_availability_zones))}"
    subnet_id = "${element(aws_subnet.private-persistence.*.id, count.index)}"
    route_table_id = "${element(aws_route_table.private-persistence.*.id, count.index)}"
}
```

### Expected Behavior

It should create the route tables (one per AZ) and associate them with the relevant subnets.

### Actual Behavior

Sometimes everything works, but intermittently, we get errors that 1 or more route tables does not exist. I'm guessing this is an eventual consistency error, so it's not predictable when/if it'll fail or for which route tables. Here's one example:

```
Error applying plan:

2 error(s) occurred:

* aws_route_table.private-persistence.2: InvalidRouteTableID.NotFound: The routeTable ID 'rtb-2d0d2f4a' does not exist
status code: 400, request id: 46696d9b-5a2f-42d3-81f3-b3c62f31a741
* aws_route_table.private-app.3: InvalidRouteTableID.NotFound: The routeTable ID 'rtb-2a0d2f4d' does not exist
status code: 400, request id: 5c5ba64d-b0e7-4493-beca-a96e7f0bfe01
```

### Steps to Reproduce

1. `terraform apply`

Note: As part of a test suite, we apply the Terraform templates in a CircleCI job that creates and destroys several dozen VPCs, each with many route tables and subnets. Most of them work fine, but roughly every other build, one of the VPCs fails due to an eventual consistency issue such as the one in this bug.

**Comment by catsby** [comment]
Hey @brikis98 – in #7156 I add a small bit of retry logic to retry the creation of the route table association and the actual routes themselves. Hopefully in large infrastructures where things are moving quite fast, this will help by retrying and allowing the eventually consistency to catch up. 

Please let me know if you feel this is insufficient, or if you have other thoughts/ideas here

Thanks!

**Comment by brikis98** [comment]
@catsby Thanks! That looks like a reasonable approach to the problem. I'd be more than happy to try it out in the next version of Terraform :)

**Comment by catsby** [comment]
Great! I merged #7156 and it will go out in the next release (v0.7.0)
